### PR TITLE
feat: add semantic-pull-request support

### PR DESCRIPTION
**Pull request by vinayakkulkarni**
- [x] Add support for [Semantic Pull Requests](https://probot.github.io/apps/semantic-pull-requests/)

### CODE REVIEW

**Comment by vinayakkulkarni** [comment]
@googlebot I signed it!

File: .github/semantic.yml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# Always validate the PR title AND all the commits
+titleAndCommits: true
+# Allows use of Merge commits (eg on github: "Merge branch 'master' into feature/ride-unicorns")
+# this is only relevant when using commitsOnly: true (or titleAndCommits: true)
+allowMergeCommits: true
```
Comment (patrickhulce) [inline .github/semantic.yml]: let's set this to `false`, we only use rebase or squash and merge

File: .github/semantic.yml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+# Always validate the PR title AND all the commits
+titleAndCommits: true
```
Comment (patrickhulce) [inline .github/semantic.yml]: let's do `titleOnly`, it seems less invasive to newcomers who will typically end up with squashed commits anyhow

Comment (vinayakkulkarni) [reply .github/semantic.yml]: - [x] Fixed in 48a65795c00df13544f0a0d60fc75c4fcb0c9f6c

File: .github/semantic.yml (hunk 1 above)
Comment (vinayakkulkarni) [reply .github/semantic.yml]: - [x] Fixed in 612db406731da7c58de5bed89c0a931f1de52355

File: .github/semantic.yml [hunk 3]
```diff
@@ -0,0 +1,14 @@
+# Always validate the PR title, and ignore the commits
+titleOnly: true
+# Always validate the PR title AND all the commits
```
Comment (patrickhulce) [inline .github/semantic.yml]: Oh, I guess we don't need these two lines if we're already using `titleOnly: true` correct?

File: .github/semantic.yml [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+titleAndCommits: false
+# You can define a list of valid scopes
+scopes:
+  - cli
+  - server
+  - utils
+  - viewer
+  - recipes
+# Allows use of Merge commits (eg on github: "Merge branch 'master' into feature/ride-unicorns")
+# this is only relevant when using commitsOnly: true (or titleAndCommits: true)
```
Comment (patrickhulce) [inline .github/semantic.yml]: does this mean we don't need this flag at all?

File: .github/semantic.yml (hunk 3 above)
Comment (vinayakkulkarni) [reply .github/semantic.yml]: According to the [configuration](https://github.com/zeke/semantic-pull-requests#configuration) docs, all these settings are optional.
I personally prefer explicitly setting all the configuration settings.

I'll remove:
```diff
- titleAndCommits: false
```

Comment (vinayakkulkarni) [reply .github/semantic.yml]: - [x] Fixed in 21b0609

File: .github/semantic.yml (hunk 4 above)
Comment (vinayakkulkarni) [reply .github/semantic.yml]: Yep, we don't need it.

- [x] Fixed in 80d2a34

**Pull request merged**